pkg/download/download.go

```diff
@@ -0,0 +1,168 @@
+package download
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/api"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/download/jsoncreator"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/download/yamlcreator"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/environment"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util/files"
+)
+
+var baseProjectName = "" //for folder naming
+var cont = 0
+
+//GetConfigs Entry point that retrieves the specified configurations from a Dynatrace tenant
+func GetConfigs(environments map[string]environment.Environment, path string, downloadSpecificAPI string) (statusCode int) {
+
+	list, err := getAPIList(downloadSpecificAPI)
+	if err != nil {
+		util.Log.Fatal("The API list contains invalid values. Run monaco command to see the available options\n" + err.Error())
+		statusCode = -1
+		return statusCode
+	}
+	downloadErrors := make(map[string]error)
+
+	for _, environment := range environments {
+
+		//download configs for each environment
+		err = downloadConfigFromEnvironment(environment, path, list)
+		if err != nil {
+			downloadErrors[environment.GetId()] = err
+		}
+	}
+	statusCode = publishErrors(downloadErrors)
+	return statusCode
+}
+
+func publishErrors(errors map[string]error) (statusCode int) {
+	for environment, err := range errors {
+		util.Log.Error("Download to %s failed with error %s\n", environment, err)
+		statusCode = -1
+	}
+	if statusCode == -1 {
+		return statusCode
+	}
+	return 0
+}
+
+//returns the list of API filter if the download specific flag is used, otherwise returns all the API's
+func getAPIList(downloadSpecificAPI string) (map[string]api.Api, error) {
+	availableApis := api.NewApis()
+	blank := strings.TrimSpace(downloadSpecificAPI) == ""
+	filterAPIList := make(map[string]api.Api)
+	if blank {
+		for _, entity := range availableApis {
+			path := transFormSpecialCasesAPIPath(entity.GetId(), entity.GetApiPath())
+			filterAPIList[entity.GetId()] = api.NewApi(entity.GetId(), path)
+		}
+		return filterAPIList, nil
+	}
+	requestedApis := strings.Split(downloadSpecificAPI, ",")
+	result := true
+	errString := ""
+
+	for _, id := range requestedApis {
+		cleanAPI := strings.TrimSpace(id)
+		isApi := api.IsApi(cleanAPI)
+		if isApi == false {
+			result = false
+			errString += errString + fmt.Sprintf(" \t - Value %s is not a valid API name.\n", cleanAPI)
+		} else {
+			filterAPI := availableApis[cleanAPI]
+			path := transFormSpecialCasesAPIPath(filterAPI.GetId(), filterAPI.GetApiPath())
+			filterAPIList[cleanAPI] = api.NewApi(filterAPI.GetId(), path)
+		}
+	}
+	if result == false {
```

Comment:
You could check for the error array length `> 0` here. Then you could get rid of the `result` variable.

Reply:
done